src/partial.rs

```diff
@@ -222,6 +226,34 @@ mod test {
         );
     }
 
+    #[test]
+    fn test_partial_subexpression_context_hash() {
+        let mut hbs = Registry::new();
+        hbs.register_template_string("one", "This is a test. {{> (x @root) name=\"fred\" }}")
+            .unwrap();
+        hbs.register_template_string("two", "Lets test {{name}}")
+            .unwrap();
+
+        hbs.register_helper(
+            "x",
+            Box::new(
+                |_: &Helper<'_, '_>,
+                 _: &Registry<'_>,
+                 _: &Context,
+                 _: &mut RenderContext<'_, '_>,
+                 out: &mut dyn Output|
+                 -> Result<(), RenderError> {
+                    out.write("two")?;
+                    Ok(())
+                },
+            ),
+        );
+        assert_eq!(
+            "This is a test. Lets test fred",
+            hbs.render("one", &0).unwrap()
+        );
+    }
```

Comment:
This test now has the same behavior as handlebars.js, as demonstrated in this playground link: https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22This%20is%20a%20test.%20%7B%7B%3E%20(x%20%40root)%20name%3D%5C%22fred%5C%22%7D%7D%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22two%22%2C%22content%22%3A%22Lets%20test%20%7B%7Bname%7D%7D%22%7D%5D%2C%22input%22%3A%22%7B%7D%22%2C%22output%22%3A%22This%20is%20a%20test.%20Lets%20test%20fred%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper(%5C%22x%5C%22%2C%20()%20%3D%3E%20%5C%22two%5C%22)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D